### ISSUE TITLE
Catima Import: Overwrite existing Entries

### ISSUE THREAD

**Issue by waffshappen** [CatimaLoyalty/Android#512]
The Catima export simply tries to insert any records it finds, skipping any existing records by failing to insert them.

Users should probbably be able to choose if to keep or overwrite old records before the Import.

**Comment by TheLastProject** [comment]
I think it would be much better if Catima popped up a dialog when it found a conflict. Something like the Nextcloud conflict dialog:

![](https://user-images.githubusercontent.com/17022173/88113886-fef04d80-cbb2-11ea-8cbc-ab40e18e3ba5.png)

Obviously we don't have date info, but we can display all fields next to each other.

**Comment by TheLastProject** [comment]
This has been improved in #1422, cards with differences are inserted as duplicates (so you at least don't lose cards when importing). But it'll still need further improvement.

**Comment by Korb** [comment]
> cards with differences are inserted as duplicates

You're right, it's far from ideal right now. Here's a situation where the app behaved unexpectedly and unpleasantly:

1. A couple of years ago, I added all of our family's discount cards to Catima.
2. I exported the zip file (700+ MB) and asked my family members to import it into their own devices, explaining that if new cards were added, they should only be added to my own device, which would be the most complete and up-to-date at any time.
3. (Presumably: over time, the instructions were forgotten, and users made some changes to their databases.)
4. Today, I exported my database and asked my relatives to add it to theirs phones.

### Expected Result

The new database overwrites the old database.

### Actual Result

Many (all?) cards were duplicated.

### Proposed Solutions

1. Add the ability to delete all cards (#2518).
2. When selecting a file for import, ask whether the user wants to replace the old database with the new one, update it (add all cards except those already added that are identical), or add to them even if they seem to match (I don't know why anyone would want to do this).
3. When creating a card, assign it a date, time, and a unique random ID that persists regardless of any changes it undergoes. Each time a card is modified, the date and time of the modification are added. When importing, the IDs are compared first. If they match for at least one card, ask the user whether to save
    - only the cards with the most recent edit date,
    - the cards on the device,
    - the cards in the imported database, or
    - both versions.

_(I did not repeat links to issues that are already linked in this issue)_
